spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2501,10 +2501,7 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
 
       case SortExec(sortOrder, _, child, _)
           if isCometOperatorEnabled(op.conf, CometConf.OPERATOR_SORT) =>
-        // TODO: Remove this constraint when we upgrade to new arrow-rs including
-        // https://github.com/apache/arrow-rs/pull/6225
-        if (child.output.length == 1 && child.output.head.dataType.isInstanceOf[StructType]) {
```

Comment:
It should be the column to sort (i.e., sort order) instead of child output. I made it wrong in #811.